snoop/common_data/models.py

```diff
@@ -36,3 +41,52 @@ class Meta:
     class PartitioningMeta:
         method = PostgresPartitioningMethod.HASH
         key = ["doc_sha3_256"]
+
+
+class NextcloudCollection(models.Model, Collection):
+    """Model for storing nextcloud collection metadata."""
+
+    name = models.CharField(max_length=256, unique=True)
+    opt = models.JSONField(null=True, blank=True)
+
+    @property
+    def webdav_user(self):
+        return self.opt.get('webdav_user', '')
+
+    @property
+    def webdav_password(self):
+        return self.opt.get('webdav_password', '')
+
+    @property
+    def webdav_url(self):
+        return self.opt.get('webdav_url', '')
+
+    @contextmanager
+    def mount_collection_webdav(self, col, nc_col):
+        """Mount a nextcloud collection via webdav.
+        """
+        subprocess.run(['mkdir', '-p',
+                        (
+                            f'{settings.SNOOP_WEBDAV_MOUNT_DIR}'
+                            f'/{self.name}/data'
+                        )
+                        ], check=True)
+        secrets_content = (
+            f'{settings.SNOOP_WEBDAV_MOUNT_DIR}'
+            f'/{self.name}/data'
+            f' {self.webdav_user} {self.webdav_password}'
+        )
+        with open('/etc/davfs2/secrets', 'a') as secrets_file:
+            secrets_file.write(f'\n{secrets_content}')
+            mount_command = (
+                f'mount -t davfs http://10.66.60.1:9972{self.webdav_url} '
```

Comment:
please move http://10.66.60.1:9972 into setting and environment variable (default '') and assert that it's set when running the mount operation.

then make another PR in node, to set the correct env var (passed through a fabio port and not a static port) so it still works in multi-node environments